aten/src/ATen/core/ivalue.cpp

```diff
@@ -940,17 +940,24 @@ TORCH_API intrusive_ptr<ivalue::Future> collectAll(
   };
 
   auto ctx = std::make_shared<Ctx>(std::move(srcs));
-  std::function<void()> func = [ctx]() {
-    if (--ctx->remaining == 0) {
-      ctx->dstFuture->markCompleted(ctx->asIvalue);
-    }
-  };
   if (ctx->srcFutures.size() == 0) {
     ctx->dstFuture->markCompleted(ctx->asIvalue);
   } else {
     auto typePtr = ctx->srcFutures.get(0)->elementType();
     for (const auto i : c10::irange(ctx->srcFutures.size())) {
-      TORCH_CHECK(i == 0 || *ctx->srcFutures.get(i)->elementType() == *typePtr);
+
+      auto fut = ctx->srcFutures.get(i);
```

Comment:
I am here just to get a bit more context about RPC. My comment is mostly nitpick. Would it be more ideal to pass the index itself or the naked pointer (as the `intrusive_ptr` is already stored in `srcFutures` as part of lambda closure)? If not, you can also use the indexing operator instead of the range-checked `get()`.

Reply:
Is the goal here to avoid the refcount bump for `intrusive_ptr`? It doesn't seem like that would be much of an issue given the use case here.